### PR TITLE
Added origin info to panel requests

### PR DESCRIPTION
**Pull request by supermassive**
What is done here:
- Added OriginInfo mojo struct for browser->bubble communication with these fields:
  - origin - Url.mojom.Origin instance to send back from bubble to browser for some action with origin.
  - origin_spec - a string to display as request source
  - e_tld_plus_one - a string with eTLD+1 
- Removed origin capturing from renderer. This should be done in browser process.
- Replaced std::string and GURL with url::Origin where we deal with origin.
  

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/common/brave_wallet.mojom
Comment (kylehickinson) [inline components/brave_wallet/common/brave_wallet.mojom:10]: FYI iOS's framework has no equivalent of `url::Origin` and so we currently do not support Obj-C code-gen for it.  I will have to look into this but at the moment this PR will not compile for iOS

File: browser/permissions/brave_ethereum_permission_context_browsertest.cc
Comment (supermassive) [inline browser/permissions/brave_ethereum_permission_context_browsertest.cc:77]: `ServeFilesFromSourceDirectory` converts arg to absolute path, so test binary could be run from everywhere.

**Comment by yrliou** [comment]
Add @SergeyZhukovsky as a reviewer for android, and @Douglashdaniel as a reviewer for front end changes.

**Review by yrliou — APPROVED** [review]
Backend changes LGTM, approved assuming CI errors (lint and storybook) will be fixed.

**Review by SergeyZhukovsky — APPROVED** [review]
Android part looks good!

**Comment by supermassive** [comment]
@SergeyZhukovsky, can you please take a look at android part once? Had to add some more changes to make java compile and I'm not an expert in it.

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:173]: I don't really get why we cannot pass `.spec()` from GURL to a mojom call and after that on c++ side convert it to GURL?

Comment (supermassive) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:173]: The idea is to switch from GURL and std::string to url::Origin to increase type-safety in origin contexts. For wallet bubble there is actually no such manual origin construction needed as it comes from browser via BraveWalletServiceObserver::OnActiveOriginChanged. Not sure if that is somehow easily applicable for android

**Comment by supermassive** [comment]
@kylehickinson @Douglashdaniel @kylehickinson @goodov 
PTAL

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:173]: ok, let it be for now!

**Review by SergeyZhukovsky — APPROVED** [review]
Android side looks ok!

**Review by kylehickinson — APPROVED** [review]
Tested the iOS changes locally

++

**Review by goodov — APPROVED** [review]
lgtm. Discussed opaque origins in person, they doesn't break everything too much, but some UI issues should be fixed.
rebase would be nice before merging.

**Pull request merged**
